src/compiler/crystal/compiler.cr

```diff
@@ -381,13 +381,13 @@ module Crystal
         end
       end
 
+      output_filename = File.expand_path(output_filename, home: true)
```

Comment:
I'd expect that calling `-o ~/foo.obj` would work.

Reply:
And it works: the shell expands it for you.

Reply:
yeah, unless it doesn't ;)